### PR TITLE
WIP: Feat/image upload

### PR DESCRIPTION
**Pull request by pedrohbl03**

### CODE REVIEW

File: src/app/(application)/layout.tsx [hunk 1]
```diff
@@ -17,14 +25,15 @@ const AppLayout = ({ children }: { children: React.ReactNode }) => {
...
-      {session?.user && isPremiumExpired && (
+      {session?.user && isPremiumExpired && isCommomUser && (
         <PremiumConversionModal
           isPremiumExpired={isPremiumExpired}
           isNewUser={session.user.subscriptionEndDate === null}
           setIsPremiumExpired={setIsPremiumExpired}
           userId={session.user.id}
         />
       )}
+      <CompleteSupplierRegister />
```
Comment (pedrohbl03) [inline src/app/(application)/layout.tsx:36]: We can make a rule to show this Modal here.

If supplier don't have a supplierLogo, Banner and description only.

File: src/app/api/auth/[...nextauth]/route.ts [hunk 2]
```diff
@@ -67,8 +67,10 @@ export const authOptions: NextAuthOptions = {
 
   callbacks: {
     async jwt({ token, user, account, trigger, session }) {
+      console.log("trigger", trigger);
```
Comment (renanpontez) [inline src/app/api/auth/[...nextauth]/route.ts:70]: +1 protein bar

File: src/app/api/auth/[...nextauth]/route.ts [hunk 3]
```diff
@@ -67,8 +67,10 @@ export const authOptions: NextAuthOptions = {
...
       if (trigger === "update") {
         token.user = session.user;
+        console.log("token", token);
```
Comment (renanpontez) [inline src/app/api/auth/[...nextauth]/route.ts:73]: +1 protein bar @pedrohbl03

File: src/components/mols/CompleteSupplierRegister/index.tsx [hunk 4]
```diff
@@ -85,6 +91,54 @@ const CompleteSupplierRegister: React.FC = () => {
...
+      ...session,
+      user: {
+        ...session?.user,
+        supplierInfo: {
+          ...session?.user.supplierInfo,
+          ...newSupplierInfo,
+        },
+      },
+    }).then(() => {
+      console.log("update");
```
Comment (renanpontez) [inline src/components/mols/CompleteSupplierRegister/index.tsx:139]: +1 protein bar @pedrohbl03

File: src/components/orgs/Coupon/List.tsx [hunk 5]
```diff
@@ -150,20 +149,35 @@ const CouponList: React.FC<ICouponListProps> = ({ supplierId }) => {
...
+            sizes="100vw"
+            style={{ width: "auto", height: "auto" }}
+            src={supplier.supplier.supplierInfo.supplierBanner}
+          />
+        )}
+      </div>
       <Link
         prefetch={true}
-        className="absolute flex items-center justify-center rounded-full top-4 left-4 cursor-pointer h-8 w-8 bg-neutral-400 opacity-75 rotate-180"
+        className="absolute flex items-center h- justify-center rounded-full top-4 left-4 cursor-pointer h-8 w-8 bg-neutral-400 opacity-75 rotate-180"
```
Comment (renanpontez) [inline src/components/orgs/Coupon/List.tsx:168]: +1 protein bar

File: src/components/orgs/Coupon/List.tsx [hunk 6]
```diff
@@ -150,20 +149,35 @@ const CouponList: React.FC<ICouponListProps> = ({ supplierId }) => {
           </ButtonThird>
         </div>
       </Modal>
-      <div className="h-40 w-full bg-gradient-to-r from-primary-lighter to-primary-main"></div>
+      <div className="h-auto pb-6 relative min-h-52 max-h-80 w-full mx-auto flex justify-center bg-cover bg-gradient-to-r from-primary-lighter to-primary-main">
+        {supplier.supplier.supplierInfo.supplierBanner && (
```
Comment (renanpontez) [inline src/components/orgs/Coupon/List.tsx:153]: change supplier.supplier to supplier.

File: src/components/orgs/Coupon/List.tsx [hunk 7]
```diff
@@ -201,12 +217,12 @@ const CouponList: React.FC<ICouponListProps> = ({ supplierId }) => {
...
-          <p className="pt-2 text-xs text-gray-400">
-            Lorem ipsum dolor sit amet consectetur adipisicing elit. Excepturi
-            vero velit quam repellendus facere ea recusandae, sapiente
-            repudiandae perspiciatis temporibus et exercitatione illum nobis
-            corrupti, sunt voluptates perferendis dicta fugiat.
-          </p>
+          {supplier.supplier.supplierInfo.supplierDescription && (
+            <p className="pt-2 text-xs text-gray-400">
+              {" "}
+              {supplier.supplier.supplierInfo.supplierDescription}{" "}
```
Comment (renanpontez) [inline src/components/orgs/Coupon/List.tsx:223]: remove spaces

File: src/types/auth/response/index.ts [hunk 8]
```diff
@@ -8,6 +10,7 @@ export interface IUser {
   active: boolean;
   isSupplier: boolean;
   isVerified: boolean;
+  supplierInfo?: ISupplierInfo | null;
```
Comment (renanpontez) [inline src/types/auth/response/index.ts:13]:
```suggestion
  supplierInfo?: ISupplierInfo;
```

File: src/types/supplier/index.ts [hunk 9]
```diff
@@ -23,13 +29,29 @@ export interface ISupplier {
       address: IAddress;
       isSupplier: boolean;
       isVerified: boolean;
+      supplierBanner: string;
+      supplierLogo: string;
+      supplierDescription: string;
```
Comment (renanpontez) [inline src/types/supplier/index.ts:34]: Change to ISupplierInfo;

File: src/types/user/index.ts [hunk 10]
```diff
@@ -13,6 +14,7 @@ export interface userLoginResponseProps {
   isVerified: boolean;
   name: string;
   role: string;
+  supplierInfo?: ISupplierInfo | null;
```
Comment (renanpontez) [inline src/types/user/index.ts:17]:
```suggestion
  supplierInfo?: ISupplierInfo;
```

**Review by renanpontez — APPROVED** [review]

**Pull request merged**